// lorem-ipsum/dist/util/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "capitalize", {
  enumerable: true,
  get: function get() {
    return _capitalize.default;
  }
});
Object.defineProperty(exports, "copyToClipboard", {
  enumerable: true,
  get: function get() {
    return _copyToClipboard.default;
  }
});
Object.defineProperty(exports, "getCopyCommand", {
  enumerable: true,
  get: function get() {
    return _getCopyCommand.default;
  }
});
Object.defineProperty(exports, "getPlatform", {
  enumerable: true,
  get: function get() {
    return _getPlatform.default;
  }
});
Object.defineProperty(exports, "getVersion", {
  enumerable: true,
  get: function get() {
    return _getVersion.default;
  }
});
Object.defineProperty(exports, "isNode", {
  enumerable: true,
  get: function get() {
    return _isNode.default;
  }
});
Object.defineProperty(exports, "isReactNative", {
  enumerable: true,
  get: function get() {
    return _isReactNative.default;
  }
});
Object.defineProperty(exports, "isSupportedPlatform", {
  enumerable: true,
  get: function get() {
    return _isSupportedPlatform.default;
  }
});
Object.defineProperty(exports, "isWindows", {
  enumerable: true,
  get: function get() {
    return _isWindows.default;
  }
});
Object.defineProperty(exports, "makeArrayOfLength", {
  enumerable: true,
  get: function get() {
    return _makeArrayOfLength.default;
  }
});
Object.defineProperty(exports, "makeArrayOfStrings", {
  enumerable: true,
  get: function get() {
    return _makeArrayOfStrings.default;
  }
});

var _capitalize = _interopRequireDefault(require("./capitalize"));

var _copyToClipboard = _interopRequireDefault(require("./copyToClipboard"));

var _getCopyCommand = _interopRequireDefault(require("./getCopyCommand"));

var _getPlatform = _interopRequireDefault(require("./getPlatform"));

var _getVersion = _interopRequireDefault(require("./getVersion"));

var _isNode = _interopRequireDefault(require("./isNode"));

var _isReactNative = _interopRequireDefault(require("./isReactNative"));

var _isSupportedPlatform = _interopRequireDefault(require("./isSupportedPlatform"));

var _isWindows = _interopRequireDefault(require("./isWindows"));

var _makeArrayOfLength = _interopRequireDefault(require("./makeArrayOfLength"));

var _makeArrayOfStrings = _interopRequireDefault(require("./makeArrayOfStrings"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
